### PR TITLE
fix windows paths in dev mode

### PR DESCRIPTION
**Pull request by MrFoxPro**
It's only for dev, I did not test it in build mode.
Please, do not forget use node path methods when working with paths

### CODE REVIEW

**Comment by MrFoxPro** [comment]
It's only for dev, I did not test it in build mode.
Please, do not forget use [node path methods](https://nodejs.org/api/path.htm) when working with paths

File: packages/start/runtime/devServer.js [hunk 1]
```diff
@@ -15,7 +15,7 @@ export function createDevHandler(viteServer) {
     try {
       if (req.url === "/favicon.ico") return;
 
-      const entry = (await viteServer.ssrLoadModule(path.resolve("/src/entry-server"))).default;
```
Comment (LXSMNSYC) [inline packages/start/runtime/devServer.js:18]: This former line is actually from Vite's SSR example, except the part for the `path.resolve`.
Screen Shot 2022-01-30 at 9 45 06 PM

What happens if we remove `path.resolve` instead?

File: packages/start/server/babel.js [hunk 2]
```diff
@@ -143,11 +144,11 @@ function transformServer({ types: t, template }) {
                   let statement = path.findParent(p => program.get("body").includes(p));
                   let serverIndex = state.servers++;
                   let hasher = state.opts.minify ? hashFn : str => str;
-                  const hash = hasher(
-                    state.filename.replace(state.opts.root, "").slice(1) + "/" + serverIndex
-                  );
+                  const fName = state.filename.replace(state.opts.root, "").slice(1)
 
-                  const route = `${INLINE_SERVER_ROUTE_PREFIX}/${hash}`;
```
Comment (LXSMNSYC) [inline packages/start/server/babel.js:150]: Wasn't this for RPC routing? I don't think we need to use `path` here.

Comment (MrFoxPro) [reply packages/start/server/babel.js:150]: hash will contain os-specific delimiters, so path will be like `a\b\c/d/e`. It is useful to use this methods to have correct path joining

File: packages/start/runtime/devServer.js (hunk 1 above)
Comment (ryansolid) [reply packages/start/runtime/devServer.js:18]: I added path resolve to handle both js and ts entry points

**Pull request merged**